reader.c

```diff
@@ -151,6 +212,10 @@ int nsq_reader_connect_to_nsqd(struct NSQReader *rdr, const char *address, int p
     struct NSQDConnection *conn;
     int rc;
 
+    if (rdr->lookupd == NULL) {
+        rdr->address = strdup(address);
+        rdr->port = port;
+    }
```

Comment:
it looks like this assumes that a reader can  only connect to a _single_ `nsqd`, it should be possible to connect to many `nsqd` directly